network/network.go

```diff
@@ -207,3 +218,49 @@ func (c *Collector) Collect(ch chan<- prometheus.Metric) {
 	scrapingElapsed := time.Since(scrapingStart)
 	c.logger.Log("level", "info", "message", "collected metrics", "scrapeID", c.scrapeID, "scrapeTime", scrapingElapsed.Seconds())
 }
+
+func (c *Collector) getNeighbours(n int, subsets []v1.EndpointSubset) ([]string, error) {
+	// Find our IP - note: this does not open a connection, due to UDP.
+	conn, err := net.Dial("udp", "8.8.8.8:80")
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	defer conn.Close()
+	ip := conn.LocalAddr().(*net.UDPAddr).IP.String()
+
+	// Get all other net-exporter IPs, and sort them.
+	addresses := []string{}
+	for _, endpointSubset := range subsets {
+		for _, address := range endpointSubset.Addresses {
+			addresses = append(addresses, address.IP)
+		}
+	}
+	sort.Strings(addresses)
+
+	// Calculate n neighbours, given our local IP and all other net-exporter IPs.
+	neighbours := c.calculateNeighbours(n, ip, addresses)
+
+	c.logger.Log("level", "info", "message", "calculated neighbours", "ip", ip, "neighbours", strings.Join(neighbours, ", "), "scrapeID", c.scrapeID)
+
+	return neighbours, nil
+}
+
+func (c *Collector) calculateNeighbours(n int, ip string, addresses []string) []string {
+	neighbours := []string{}
+
+	if n > len(addresses) {
+		n = len(addresses)
```

Comment:
```suggestion
		return addresses
```

Reply:
This is not good actually. The node would check itself with that.

Reply:
This suggestion is legit in light of this comment https://github.com/giantswarm/net-exporter/pull/66#discussion_r340518481